src/components/main-view/main-view.jsx

```diff
@@ -3,19 +3,27 @@ import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
 import { LoginView } from "../login-view/login-view";
 import { SignupView } from "../signup-view/signup-view";
+import { ProfileView } from "../profile-view/profile-view";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
-import Button from "react-bootstrap/Button";
 import "./main-view.scss";
+import {
+  Routes,
+  Route,
+  Navigate,
+  BrowserRouter,
+  useParams,
+} from "react-router-dom";
+import { NavigationBar } from "../navigation-bar/navigation-bar";
+import { Container } from "react-bootstrap";
 
 export const MainView = () => {
   const storedUser = JSON.parse(localStorage.getItem("user"));
   const storedToken = localStorage.getItem("token");
   const [token, setToken] = useState(storedToken ? storedToken : null);
   const [user, setUser] = useState(storedUser ? storedUser : null);
   const [movies, setMovies] = useState([]);
-  const [selectedMovie, setSelectedMovie] = useState(null);
-
+  const { title } = useParams();
   // connect app to api with hook
   useEffect(() => {
```

Comment:
This is total valid code and I'll show you another way: Typically with fetching data, you'll have some sort of data layer (backend, GQL, Redux) and you'll do your fetching and error handling there. Short of that. the next best practice is custom hooks. Calling the data in the hook separates your concerns of fetching the data, from the presentation layer (the component). This makes your components cleaner and less coupled to the logic. 

For example, you're hook would look like:
```
import { useState, useEffect } from 'react';

const useFetchMovies = (token) => {
  const [movies, setMovies] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(null);

  useEffect(() => {
    if (!token) {
      return;
    }

    setIsLoading(true);

    fetch("https://ajmovies-fc7e7627ec3d.herokuapp.com/movies", {
      headers: { Authorization: `Bearer ${token}` },
    })
      .then((response) => {
        if (!response.ok) {
          throw new Error('Network response was not ok');
        }
        return response.json();
      })
      .then((movies) => {
        setMovies(movies);
        setIsLoading(false);
      })
      .catch((error) => {
        setError(error);
        setIsLoading(false);
      });
  }, [token]);

  return { movies, isLoading, error };
};

```

And your component would now look like this:
```
export const MainView = () => {
  // ... (other state and logic remains the same)

  const { movies, isLoading, error } = useFetchMovies(token); // <---- THE NEW HOOK

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error.message}</div>;
  }

  // ... (rest of your component logic remains the same)
};
```

Now the great thing is,  when the hook fetches the data, `movies` will be updated and you can wait for the changes in this component with a useEffect

`useEffect(() => {
   // update something else etc..
}, [movies]);`

This is a good practice because when you get into testing, and having to scale a codebase, this makes everything very predictable and consistent. It also makes it easy to change data or the component later. Imagine if someone said "hey now make the component a nested component in this other thing" you'd have to rip out all the logic and move it around. With this system, you just move the call of the hook   `const { movies, isLoading, error } = useFetchMovies(token);` to the new place.